recipes/lockscreen.md

```diff
@@ -0,0 +1,194 @@
+# Lockscreen
+
+_By [adamlubek](https://github.com/adamlubek)_
+
+This recipe demonstrates RxJs implementation of lockscreen functionality (known for example from smartphones).
+
+<div class="ua-ad"><a href="https://ultimatecourses.com/?ref=76683_kee7y7vk"><img src="https://ultimatecourses.com/assets/img/banners/uc-leader.svg" style="width:100%;max-width:100%"></a></div>
+
+### Example Code
+
+(
+[StackBlitz](https://stackblitz.com/edit/rxjs-lockscreen?file=index.ts&devtoolsheight=30)
+)
+
+#### index.ts
+```js
+/*
+  Use mouse to 'swipe' across the lock pad (hold mouse button and swipe :) ).
+  Pad will turn green if password is correct or red if password is incorrect.
+  You can set password to whatever sequence you like.
+*/
+
+// RxJS v6+
+import { from, fromEvent, Subject, merge } from 'rxjs';
+import { switchMap, takeUntil, repeat, tap, map, throttleTime, distinctUntilChanged, filter, toArray, sequenceEqual, pluck } from 'rxjs/operators';
+import { pads, resetPasswordPad, setResult, setTouched, updateCurrentPassword } from './dom-updater';
+
+const sub = new Subject();
+const expectedPasswordUpdate$ = fromEvent(document.getElementById('expectedPassword'), 'keyup')
+  .pipe(
+    map((e: any) => e.target.value),
+    tap(pass => sub.next(pass.split('').map(e => parseInt(e))))
+  );
+let expectedPassword = [1, 2, 5, 2];
+const expectedPassword$ = sub.pipe(tap((v: any) => expectedPassword = v));
+
+const checkIfPasswordMatch = password => from(password).pipe(sequenceEqual(from(expectedPassword)));
+
+const actualPassword$ = fromEvent(document, 'mousedown')
+  .pipe(
+    tap(resetPasswordPad),
```

Comment:
Would you be able to add comments before each operator - or at least the primary actions - to explain why you are using it. I think for more complicated examples like this it will be hugely beneficial to users newer to RxJS (or a particular operator) to step through your thought process.

For instance:

```
// switchMap so we aren't maintaining old listeners from previous drags
switchMap(_ => fromEvent(document, 'mousemove'))
// once mouse is up drag is over so we can end this chain
takeUntil(fromEvent(document, 'mouseup'))
...etc
```

Reply:
I just refactored the code to make main piece of functionality more declarative, flow like sentences. Do you think that's enough or would you still see a need for additional comments?

Reply:
I think it's good, I'm going to merge. :)

Reply:
sweet :) I don't mind adding more comments though if you think more are required, just let me know :)

Reply:
The more the better in my opinion, particularly on operators that are less used (retry), or are key to the example. Some of our visitors will be familiar without the additional explanations, but it will be gold to those who are not!

Reply:
I'm just not a fan of comments generally, prefer self explanatory code. That said though, learning materials are different so you're right. I'll see what can be added :)